src/run/mod.rs

```diff
@@ -0,0 +1,48 @@
+use args::RunSubcommands;
+
+use crate::{
+    build::ensure_web_setup,
+    external_cli::{cargo, wasm_bindgen, CommandHelpers},
+    manifest::package_name,
+};
+
+pub use self::args::RunArgs;
+
+mod args;
+mod serve;
+
+pub fn run(args: &RunArgs) -> anyhow::Result<()> {
+    let cargo_args = args.cargo_args_builder();
+
+    if let Some(RunSubcommands::Web(web_args)) = &args.subcommand {
+        ensure_web_setup()?;
+
+        // If targeting the web, run a web server with the WASM build
+        println!("Building for WASM...");
+        cargo::build::command().args(cargo_args).ensure_status()?;
+
+        println!("Bundling for the web...");
+        wasm_bindgen::bundle(&package_name()?, args.profile())?;
+
+        let port = web_args.port;
+        let url = format!("http://127.0.0.1:{port}");
+
+        // Serving the app is blocking, so we open the page first
+        if web_args.do_open {
+            if webbrowser::open(&url).is_err() {
+                println!("Failed to open the browser automatically, open the app on <{url}>");
+            } else {
+                println!("Your app is running on <{url}>");
+            }
```

Comment:
(Nit)

1. Since we're handling both the `Ok` and `Err` variants, I would prefer to use a `match` statement instead. We can even log the error when this happens, so users can diagnose the issue.
2. "open the app on <http://127.0.0.1:4000>" sounds weird when I say it in my head. Since "<...>" is an address, not a place, I think "open the app _at_ <...>" sounds better.

Combining these two together, we get:

```suggestion
            match webbrowser::open(&url) {
                Ok(()) => println!("Your app is running at <{url}>!"),
                Err(error) => println!("Failed to open the browser automatically, open the app at <{url}>. (Error: {error})"),
            }
```